packages/react-components/src/components/chart/store/contextDataStreams.ts

```diff
@@ -0,0 +1,39 @@
+import { DataStream } from '@iot-app-kit/core';
+import { StateCreator } from 'zustand';
+
+export interface DataStreamsData {
+  highlightedDataStreams: DataStream[];
+  hiddenDataStreams: DataStream[];
+}
+
+export interface DataStreamsState extends DataStreamsData {
+  highlightDataStream: (datastream?: DataStream) => void;
```

Comment:
Nice! i like that you are not re-using the same action to perform 2 dofferent things